charts/metering-ansible-operator/templates/crds/report.crd.yaml

```diff
@@ -35,3 +35,153 @@ spec:
   - name: Age
     type: date
     JSONPath: .metadata.creationTimestamp
+  validation:
+    openAPIV3Schema:
+      type: object
+      required:
+      - spec
+      properties:
+        # note: see https://github.com/operator-framework/operator-metering/blob/master/Documentation/reports.md for more information
+        # note: apiVersion, kind, metadata is implicitly validated by API server
+        spec:
+          type: object
+          required:
+          - query
+          properties:
+            # there's no way to enforce what reportqueries get passed through the `name` field (besides the default ones using the `enum` array keyword)
+            # due to roll-up reports, custom reportqueries, etc. We already validate that the ReportQuery exists in reports.go
+            query:
+              type: string
+              minLength: 1
+
+            reportingStart:
+              type: string
+              format: date-time
+
+            reportingEnd:
+              type: string
+              format: date-time
+
+            runImmediately:
+              type: boolean
+
+            inputs:
+              type: array
+              minItems: 1
+              items:
+                type: object
+                required:
+                - name
+                - value
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  value: {}
+
+            schedule:
+              type: object
+              required:
+              - period
+              properties:
+                period:
+                  type: string
+                  minLength: 1
+                  enum:
+                  - hourly
+                  - daily
+                  - weekly
+                  - monthly
+                  - cron
+
+                hourly:
+                  type: object
+                  properties:
+                    minute:
+                      type: integer
+                      minimum: 0
+                      maximum: 59
+                    hour:
+                      type: integer
+                      minimum: 0
+                      maximum: 23
+
+                daily:
+                  type: object
+                  properties:
+                    second:
+                      type: integer
+                      minimum: 0
+                      maximum: 60
+                    minute:
+                      type: integer
+                      minimum: 0
+                      maximum: 59
+                    hour:
+                      type: integer
+                      minimum: 0
+                      maximum: 23
+
+                weekly:
+                  type: object
+                  properties:
+                    dayofWeek:
+                      type: string
```

Comment:
This could have an enum validation for it. Checkout `convertDayOfWeek` in `pkg/operator/reports.go`